### ISSUE TITLE
cloudinary.v2.api.resources_by_tag() -- (tags, callback, options) parameter options incorrect

### ISSUE THREAD

**Issue by leonvisscher** [cloudinary/cloudinary_npm#272]
Documentation suggests you should do:
`cloudinary.v2.api.resources_by_tag(tags, callback, options)`
But this gives an error: `Uncaught Exception:  TypeError: callback is not a function`

When i do:
`cloudinary.v2.api.resources_by_tag(tags, options, callback)`

Everything works fine.

**Comment by aditimadan-Cloudinary** [comment]
Can you please share the link to the documentation and I will let the doc team know of the update.

**Comment by jbjhjm** [comment]
Experiencing the same issue using v2.uploader.upload_large_stream. 
Documentation might still list v1 arguments.

Investigation lead me to the pretty weird "v1_adapters" method called in lib/v2/uploader. 
It tries to dynamically generate functions with a standardized argument interface taking the specified number of arguments plus, as last argument, an optional callback.

This concept is applied to all the listed below methods, so all of these have to be called with  (arg1, arg2, argX, callback).

```js
v1_adapters(exports, uploader, {
  unsigned_upload_stream: 1,
  upload_stream: 0,
  unsigned_upload: 2,
  upload: 1,
  upload_large_part: 0,
  upload_large: 1,
  upload_chunked: 1,
  upload_chunked_stream: 0,
  explicit: 1,
  destroy: 1,
  rename: 2,
  text: 1,
  generate_sprite: 1,
  multi: 1,
  explode: 1,
  add_tag: 2,
  remove_tag: 2,
  remove_all_tags: 1,
  add_context: 2,
  remove_all_context: 1,
  replace_tag: 2,
  create_archive: 0,
  create_zip: 0
});
```

This dynamic adapter makes it pretty hard to follow what's actually required to make it work.
IMO, hardcoded adapter methods would be a lot easier and clearer to follow and to understand.

**Comment by cloudinaraz** [comment]
@leonvisscher can you please direct us to where in the documentation you still see the callback being the second argument for the `resources_by_tag()` method or any other method?
We tried looking it up but so far couldn't find any occurrence of what you were referring to.

@jbjhjm thank you for your feedback. The APIs will be refactored in the next major release.

**Comment by strausr** [comment]
Closing this issue due to the time elapsed. Please feel free to either re-open the issue, contact our support at http://support.cloudinary.com or create a new ticket if you have any additional issues.
